hazelcast/src/main/java/com/hazelcast/collection/impl/set/SetContainer.java

```diff
@@ -75,7 +77,15 @@ public SetConfig getConfig() {
     public Set<CollectionItem> getCollection() {
         if (itemSet == null) {
             if (itemMap != null && !itemMap.isEmpty()) {
-                itemSet = new HashSet<CollectionItem>(itemMap.values());
+                itemSet = createHashSet(itemMap.size());
+                long maxItemId = Long.MIN_VALUE;
+                for (CollectionItem collectionItem : itemMap.values()) {
+                    if (collectionItem.getItemId() > maxItemId) {
+                        maxItemId = collectionItem.getItemId();
+                    }
+                    itemSet.add(collectionItem);
+                }
+                setId(maxItemId + ID_PROMOTION_OFFSET);
```

Comment:
We enter this branch, when backups of `ISet`  items are promoted. The backups are stored in the `itemMap` and are used to populate the `itemSet`. The issue was to not call `setId()` with the maximum `itemId` of the existing items.